NebulaClient/MonoBehaviours/MultiplayerSession.cs

```diff
@@ -88,11 +99,24 @@ void Update()
         private void OnJoinSessionConfirmed(JoinSessionConfirmed packet)
         {
             PlayerManager.SetLocalPlayer(packet.LocalPlayerId);
+            if (statusBox != null)
+            {
+                statusBox.FadeOut();
+                statusBox = null;
+            }
         }
 
         private void OnRemotePlayerJoined(RemotePlayerJoined packet)
         {
             PlayerManager.AddRemotePlayer(packet.PlayerId);
+            statusBox = UIMessageBox.Show("A new player is joining!", "A new player is joining, please wait while they load in", null, 0);
+            GameMain.Pause();
+
+            if(PlayerManager.WeAreMainPlayer)
+            {
+                GameSave.SaveCurrentGame("MPSYNCSTATE");
```

Comment:
I think we should use the `DSPGame.LoadFile` property to get the current save file name instead of creating our own. Otherwise, we won't be able to play 2 differents multiplayer games since when you will load the second game, it will override the first one.

For now since the `DSPGame.LoadFile` is empty when we start a new game, we could save it with a hardcoded  timesptamp.